packages/onboarding/src/step-navigation-link/index.tsx

```diff
@@ -59,7 +59,9 @@ const StepNavigationLink: React.FC< Props > = ( {
 			primary={ primary }
 			borderless={ borderless }
 			className={ buttonClasses }
-			onClick={ onClick }
+			// Use onMouseDown here to prevent onClick blocked by onBlur
+			// if any exists in adjacent elements
+			onMouseDown={ onClick }
```

Comment:
Out of curiosity, which browsers did you test this in?

Reply:
On Chrome. Do you want me to test it on others like Firefox for example? (I'm on linux, so Safari is not an option for me.)

Reply:
What about mobile?

Reply:
To expand on my thought process: semantically, it doesn't feel correct to change this from `onClick` to `onMouseDown`. It seems like a bug that some `onBlur` preventing `onClick`...

Reply:
There is indeed an `onBlur` in a neighbor element. Leaving it as is, means that we can't use an onBlur that triggers changes in state, and even if we fix that, there is a possibility that this bug will reappear. I'll check mobile and if doesn't work, I'll try to change the `onBlur` function. However, this onBlur is deep inside the StepContainer and I'm not sure what ll be affected by doing that.

Reply:
Tested it on Android on Chrome and Firefox. Seems that works as expected. If you or @sejas can give this a try on Safari, it would be great.

Reply:
It works on mobile 👌 .
It would be great if we could use the `onClick` and move the button to the left like in the importer flow.
I know it's not an easy task because our stepper and the import flow use different components, so I'm fine creating a new issue to improve this.

| <img width="1279" alt="Screenshot 2023-01-24 at 13 30 05" src="https://user-images.githubusercontent.com/779993/214307500-0da5d1e0-40fb-4c83-bb3a-c7d333b2e6e8.png"> |
|---|
| https://wordpress.com/start/import/domains |

Reply:
@kozer Want to share the bug you've found on calypsop2 so we can get broader input on how we might solve it?

Reply:
For future reference: p4TIVU-aoH-p2

Reply:
@danielbachhuber , @sejas I'll wait for the outcome of this p2 post to see how I'll proceed from here.
In the meantime, I pushed some changes, changing what I had pushed originally with the second solution I propose in this post. 
It restores the functionality as it was + the `onMouseDown` with `preventDefault`, so we won't have any unexpected behavior as we could have with `onMouseDown`